File: nemonic_sdk_flutter/nemonic_sdk/android/src/main/java/com/mangoslab/nemonicsdk/nemonic_sdk/NemonicSdkScanController.java
package com.mangoslab.nemonicsdk.nemonic_sdk;

import android.content.Context;
import android.util.Log;

import com.mangoslab.nemonicsdk.INPrinterScanControllerCallback;
import com.mangoslab.nemonicsdk.NPrinter;
import com.mangoslab.nemonicsdk.NPrinterScanController;

import java.util.HashMap;
import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;

public class NemonicSdkScanController implements INPrinterScanControllerCallback {
    private static final String TAG = "NemonicSdkScanController";
    private MethodChannel channel;
    private NPrinterScanController printerScanController;

    public NemonicSdkScanController(MethodChannel channel, Context context) {
        this.channel = channel;
        printerScanController = new NPrinterScanController(context, this);
    }

    public int startScan() {
        return printerScanController.startScan();
    }

    public void stopScan() {
        printerScanController.stopScan();
    }

    @Override
    public void deviceFound(NPrinter printer) {
        // Log.d(TAG, "deviceFound name = %s", printer.getName());
        Map<String, Object> data = new HashMap<String, Object>();

        data.put("name", printer.getName());
        data.put("macAddress", printer.getMacAddress());
        data.put("type", printer.getType().getValue());

        channel.invokeMethod("deviceFound", data);
    }
}

File: nemonic_sdk_flutter/nemonic_sdk/ios/Classes/NemonicSdkScanController.swift
import Flutter
import NemonicSdk

class NemonicSdkScanController: NSObject {
    private var channel: FlutterMethodChannel
    private var printerScanController: NPrinterScanController!

    init(channel: FlutterMethodChannel) {
        self.channel = channel
        super.init()
        printerScanController = NPrinterScanController(self)
    }

    public func startScan() -> Int {
        return printerScanController.startScan()
    }

    public func stopScan() {
        return printerScanController.stopScan()
    }
}

extension NemonicSdkScanController: NPrinterScanControllerDelegate {
    public func deviceFound(_ printer: NPrinter) {
        let data: [String: Any] = [
            "name": printer.getName(),
            "macAddress": printer.getMacAddress(),
            "type": printer.getType().rawValue
        ]

        channel.invokeMethod("deviceFound", arguments: data)
    }
}
